Content.Shared/Body/Systems/SharedBodySystem.Body.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System.Linq;
+using Content.Shared.Body.Components;
+using Content.Shared.Body.Organ;
+using Content.Shared.Body.Part;
+using Content.Shared.Body.Prototypes;
+using Content.Shared.Coordinates;
+using Robust.Shared.Containers;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Body.Systems;
+
+public partial class SharedBodySystem
+{
+    public void InitializeBody()
+    {
+        SubscribeLocalEvent<BodyComponent, MapInitEvent>(OnBodyMapInit);
+        SubscribeLocalEvent<BodyComponent, ComponentInit>(OnBodyInit);
+
+        SubscribeLocalEvent<BodyComponent, ComponentGetState>(OnBodyGetState);
+        SubscribeLocalEvent<BodyComponent, ComponentHandleState>(OnBodyHandleState);
+    }
+
+    private void OnBodyMapInit(EntityUid bodyId, BodyComponent body, MapInitEvent args)
+    {
+        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+        if (body.Prototype == null || body.Root != null)
+            return;
+
+        var prototype = _prototypes.Index<BodyPrototype>(body.Prototype);
+        InitBody(body, prototype);
+    }
+
+    private void OnBodyInit(EntityUid bodyId, BodyComponent body, ComponentInit args)
+    {
+        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+        if (body.Prototype == null || body.Root != null)
+            return;
+
+        var prototype = _prototypes.Index<BodyPrototype>(body.Prototype);
+        InitBody(body, prototype);
+    }
+
+    private void OnBodyGetState(EntityUid uid, BodyComponent body, ref ComponentGetState args)
+    {
+        args.State = new BodyComponentState(body.Root, body.GibSound);
+    }
+
+    private void OnBodyHandleState(EntityUid uid, BodyComponent body, ref ComponentHandleState args)
+    {
+        if (args.Current is not BodyComponentState state)
+            return;
+
+        body.Root = state.Root;
+        body.GibSound = state.GibSound;
+    }
+
+    private void InitBody(BodyComponent body, BodyPrototype prototype)
+    {
+        var root = prototype.Slots[prototype.Root];
+        var bodyId = Spawn(root.Part, body.Owner.ToCoordinates());
+        var partComponent = Comp<BodyPartComponent>(bodyId);
+        var slot = new BodyPartSlot(root.Part, body.Owner, partComponent.PartType);
+        body.Root = slot;
+        partComponent.Body = bodyId;
+
+        _containers.EnsureContainer<Container>(body.Owner, BodyContainerId);
```

Comment:
I thought about doing that, but I think that would make the current load-save test that checks if components were changed harder, since right now parts have an empty container in YAML to make it pass, wouldn't they need an unknown number of container slots instead, making the test impossible to fix for body parts?

Reply:
@ElectroJr